cmd/stopper.go

```diff
@@ -0,0 +1,42 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+)
+
+// Killer kills some resource and performs cleanup
+type Killer func() error
+
+// NewApplicationStopper invokes all killers and stops application
+func NewApplicationStopper(killers ...Killer) func() {
+	return newStopper(os.Exit, killers...)
+}
+
+type exitter func(code int)
+
+func newStopper(exit exitter, killers ...Killer) func() {
```

Comment:
to me it's confusing as there are three terms introduced in this PR -- `Killer` _Stopper_ and *Exiter*. Thats in addition to existing `Terminator`.
I just really want to understand heuristic differences of these terms. thanks.

Reply:
- `Killer` is a callback which tries to kill a command, but can fail (can return error).
- `Stopper` is a callback which stops the application by invoking multiple killers and exiting application using `exiter`.
- `exiter` is just a local interface (that's why it's private) for dependency-injecting `os.exit` to be able to test stopper.

Reply:
thanks, if I understand correctly, `Stopper == makeUnstoppable(Killer)`
Stopper in this case takes the role of ultimate `Destroyer` -- it destroys the app no matter what.
and `exiter` is the tool that he uses to destroy. bazooka? xD

Reply:
exaclty 👍